src/Standards/Squiz/Tests/Scope/StaticThisUsageUnitTest.inc

```diff
@@ -57,22 +57,61 @@ class MyClass
 
     public function getAnonymousClass() {
         return new class() {
-			public static function something() {
-				$this->doSomething();
+            public static function something() {
+                $this->doSomething();
             }
         };
     }
 }
 
 trait MyTrait {
-	public static function myFunc() {
-		$this->doSomething();
-	}
+    public static function myFunc() {
+        $this->doSomething();
+    }
 }
 
 $b = new class()
 {
-	public static function myFunc() {
-		$this->doSomething();
-	}
+    public static function myFunc() {
+        $this->doSomething();
+    }
+
+    public static function other() {
+        return fn () => $this->name;
+    }
+
+    public static function anonClassUseThis() {
+        return new class($this) {
```

Comment:
Currently, when we find anonymous class we are skipping till scope_closer, but we can use $this in params, so we really shouldn't skip it 'yet'.

The check should be more complicated here - please see also example in lines 98-110.
Errors should be reported only in line 84 and 99, not in 93 or 106.

I've implemented "similar" sniff here: 
https://github.com/webimpress/coding-standard/blob/develop/src/WebimpressCodingStandard/Sniffs/PHP/StaticCallbackSniff.php
which checks if closure can be defined as static or not.